clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -1804,9 +1832,10 @@ struct CounterCoverageMappingBuilder
 
     Counter LoopCount =
         addCounters(ParentCount, BackedgeCount, BC.ContinueCount);
-    Counter OutCount =
-        addCounters(BC.BreakCount, subtractCounters(LoopCount, BodyCount));
-    if (OutCount != ParentCount) {
+    auto [ExecCount, ExitCount] = getBranchCounterPair(S, LoopCount);
+    assert(ExecCount.isZero() || ExecCount == BodyCount);
+    Counter OutCount = addCounters(BC.BreakCount, ExitCount);
```

Comment:
this looks like it ought to be a helper function as well

Reply:
Could you elaborate? I didn't want to try refactoring too much here.

Reply:
I can't remember what I wanted here.